phoenix-core/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java

```diff
@@ -90,128 +90,157 @@ public CDCGlobalIndexRegionScanner(final RegionScanner innerScanner,
         super(innerScanner, region, scan, env, dataTableScan, tupleProjector, indexMaintainer,
                 viewConstants, ptr, pageSizeMs, queryLimit);
         CDCUtil.initForRawScan(dataTableScan);
-        dataColQualNameMap = ScanUtil.deserializeColumnQualifierToNameMap(
-                scan.getAttribute(DATA_COL_QUALIFIER_TO_NAME_MAP));
-        dataColQualTypeMap = ScanUtil.deserializeColumnQualifierToTypeMap(
-                scan.getAttribute(DATA_COL_QUALIFIER_TO_TYPE_MAP));
+        cdcDataTableInfo = CDCTableInfo.createFromProto(CDCInfoProtos.CDCTableDef
+                .parseFrom(scan.getAttribute(CDC_DATA_TABLE_DEF)));
+        try {
+            cdcChangeScopeSet = CDCUtil.makeChangeScopeEnumsFromString(
+                    cdcDataTableInfo.getCdcIncludeScopes());
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
     }
 
     @Override
     protected Scan prepareDataTableScan(Collection<byte[]> dataRowKeys) throws IOException {
+        //TODO: Get Timerange from the start row and end row of the index scan object
+        // and set it in the datatable scan object.
+//        if (scan.getStartRow().length == 8) {
+//            startTimeRange = PLong.INSTANCE.getCodec().decodeLong(
+//              scan.getStartRow(), 0, SortOrder.getDefault());
+//        }
+//        if (scan.getStopRow().length == 8) {
+//            stopTimeRange = PLong.INSTANCE.getCodec().decodeLong(
+//              scan.getStopRow(), 0, SortOrder.getDefault());
+//        }
         return CDCUtil.initForRawScan(prepareDataTableScan(dataRowKeys, true));
     }
 
     protected boolean getNextCoveredIndexRow(List<Cell> result) throws IOException {
         if (indexRowIterator.hasNext()) {
             List<Cell> indexRow = indexRowIterator.next();
-            for (Cell c: indexRow) {
-                if (c.getType() == Cell.Type.Put) {
-                    result.add(c);
-                }
+            // firstCell: Picking the earliest cell in the index row so that
```

Comment:
Please add this comment : A given CDC index row can have either one put cell, or a delete marker (Delete Family) and a put cell due to DML operations. When a data table column is dropped by a DDL operation and a second put cell with a higher timestamp is inserted to an index row. The timestamp of this put cell matches with the timestamp of the corresponding DeleteColumn mutation on the data table row. This DDL mutation will not show up in the CDC stream.  Index row delete marker timestamp is always higher than or equal to the put cell timestamp. The delete markers are for maintaining the index rows and can be ignored. The oldest put cell will be used to point back to the data table row version. The timestamp of the put cell is used the identify the corresponding data table mutation which can a put or delete mutation. This is the reason we only consider the oldest cell (i.e. first cell) here.